### PR TITLE
feat: add gossipsub bindings

### PR DESCRIPTION
**Pull request by MegaRedHand**

### CODE REVIEW

File: test/unit/libp2p_test.exs [hunk 1]
```diff
@@ -203,4 +203,75 @@ defmodule Libp2pTest do
...
+
+    {:ok, topic_sender} = Libp2p.pub_sub_join(gsub_sender, topic)
+    {:ok, topic_recver} = Libp2p.pub_sub_join(gsub_recver, topic)
+
+    {:ok, sub_recver} = Libp2p.topic_subscribe(topic_recver)
+
+    spawn(fn ->
+      {:ok, message} = Libp2p.subscription_next(sub_recver)
+      data = Libp2p.message_data(message)
+      IO.puts("received: #{data}")
```
Comment (mpaulucci) [inline test/unit/libp2p_test.exs:266]: 🥇

File: lib/libp2p/libp2p.ex [hunk 2]
```diff
@@ -186,35 +213,66 @@ defmodule Libp2p do
...
+  def topic_publish(_topic, _data), do: :erlang.nif_error(:not_implemented)
+
+  @doc """
+  Gets next message of given subscription.
+  """
+  @spec subscription_next(subscription) :: {:ok, message} | error
+  def subscription_next(_subscription), do: :erlang.nif_error(:not_implemented)
+
+  @doc """
+  Gets the application data from a message.
```
Comment (mpaulucci) [inline lib/libp2p/libp2p.ex:274]: what is the application data from a message?

**Review by mpaulucci — APPROVED** [review]
Awesome!

File: lib/libp2p/libp2p.ex (hunk 2 above)
Comment (MegaRedHand) [reply lib/libp2p/libp2p.ex:274]: It's the raw data sent by a peer, without any of the headers added by pubsub

**Pull request merged**